src/tools/cephfs/cephfs-shell

```diff
@@ -1113,33 +1114,47 @@ sub-directories, files')
 
     du_parser = argparse.ArgumentParser(
         description='Disk Usage of a Directory')
-    du_parser.add_argument(
-        'dirs', type=str, help='Name of the directory.', nargs='?', default='.')
+    du_parser.add_argument('paths', type=str, help='Name of the directory.',
+                           nargs='*', default='.')
     du_parser.add_argument('-r', action='store_true',
                            help='Recursive Disk usage of all directories.')
 
     @with_argparser(du_parser)
     def do_du(self, args):
         """
-        Disk Usage of a Directory
+        Print disk usage of a given path(s).
         """
-        if args.dirs == '':
-            args.dirs = cephfs.getcwd().decode('utf-8')
-        for dir_ in args.dirs:
+        def print_disk_usage(files):
+            if isinstance(files, str):
+                files = (files, )
+
+            for f in files:
+                try:
+                    f = os.path.normpath(f)
+                    if f[0] is '/':
+                        f = '.' + f
+
+                    st = cephfs.lstat(f)
+
+                    if stat.S_ISDIR(st.st_mode):
+                        dusage = int(cephfs.getxattr(to_bytes(f),
+                            'ceph.dir.rbytes').decode('utf-8'))
+                    elif stat.S_ISREG(st.st_mode) or stat.S_ISLNK(st.st_mode):
+                        dusage = st.st_size
+
+                    self.poutput('{:10s} {}'.format(humansize(dusage), f))
+                    # to avoid cases where dusage from previous iterations
+                    # ends up being reused.
+                    del dusage
```

Comment:
I'd prefer that you just add an `else: dusage = None` instead.

Reply:
Cool. Assigning `None` is definitely better than deleting it. But adding it to`else` conveys a little less that we want variable refreshed (if that's right word) at the end of (every) iteration. 

Perhaps, just `s/del dusage/dusage = None/` would be better? I am going with that for now.

Reply:
Well, the more pertinent issue is that the code is printing a `None` size for a directory entry. The `if ... elif ...` code leaves the reader wondering: is it an error that `dusage = None`? Should that never happen? Setting `dusage` at the end of the code block to affect future iterations of the loop is a code smell. (The unfortunate situation that Python has function-level scoping for variables is not a behavior to rely on in code because it's confusing.)

Reply:
Perhaps, add a check like `if dusage is not None` before `self.poutput('{:10s} {}'.format(humansize(dusage), f)`? And/or perhaps replace `dusage = None` by `del dusage`? A check before `poutput` would be required any way.

Reply:
Moved `dusage = None` from the end to the `else`. Wouldn't this lead to a bug? Now the entire if-elif-else can be read as "if file is a dir, reg file or link, print disk usage otherwise don't."

Perhaps, the `elif` part should be rewritten under `else` and `elif` part should be removed altogether.

Reply:
I've written this part. Please take a look.

Reply:
LGTM